src/test/ui/consts/issue-52432.stderr

```diff
@@ -16,7 +16,13 @@ error[E0282]: type annotations needed
 LL |     [(); &(static |x| {}) as *const _ as usize];
    |                    ^ consider giving this closure parameter a type
 
-error: aborting due to 3 previous errors
+error[E0080]: evaluation of constant value failed
+  --> $DIR/issue-52432.rs:7:10
+   |
+LL |     [(); &(static || {}) as *const _ as usize];
+   |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ "pointer-to-integer cast" needs an rfc before being allowed inside constants
```

Comment:
The "const_raw_ptr_to_usize_cast" feature gate is enabled for this test.

Reply:
yes, but that only changes static checks. Const eval can still bail out because the runtime value is a real pointer and not an integer pointer. This will become an integer pointer in https://github.com/rust-lang/rust/pull/67000